catalog/registry/catalog-registry-federationadminimpl/src/main/java/ddf/catalog/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -0,0 +1,298 @@
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE,
+                true));
+        return federationAdminService.addRegistryEntry(metacard);
+    }
+
+    @Override
+    public void updateLocalEntry(Map<String, Object> registryObjectMap)
+            throws FederationAdminException {
+        if (MapUtils.isEmpty(registryObjectMap)) {
+            throw new FederationAdminException(
+                    "Error updating local registry entry. Null map provided.");
+        }
+        RegistryPackageType registryPackage =
+                RegistryPackageWebConverter.getRegistryPackageFromWebMap(registryObjectMap);
+
+        if (registryPackage == null) {
+            String message =
+                    "Error updating local registry entry. Couldn't convert registry map to a registry package.";
+            LOGGER.error("{} Registry Map: {}", message, registryObjectMap);
+            throw new FederationAdminException(message);
+        }
+
+        List<Metacard> existingMetacards =
+                federationAdminService.getLocalRegistryMetacardsByRegistryIds(Collections.singletonList(
+                        registryPackage.getId()));
+
+        if (CollectionUtils.isEmpty(existingMetacards)) {
+            String message = "Error updating local registry entry. Registry metacard not found.";
+            LOGGER.error("{} Registry ID: {}", message, registryPackage.getId());
+            throw new FederationAdminException(message);
+        }
+
+        if (existingMetacards.size() > 1) {
+            String message =
+                    "Error updating local registry entry. Multiple registry metacards found.";
+
+            List<String> metacardIds = new ArrayList<>();
+            metacardIds.addAll(existingMetacards.stream()
+                    .map(Metacard::getId)
+                    .collect(Collectors.toList()));
+            LOGGER.error("{} Matching registry metacard ids: {}", message, metacardIds);
+
+            throw new FederationAdminException(message);
+        }
+        Metacard existingMetacard = existingMetacards.get(0);
+
+        Metacard updateMetacard = getRegistryMetacardFromRegistryPackage(registryPackage);
+        updateMetacard.setAttribute(new AttributeImpl(Metacard.ID, existingMetacard.getId()));
+
+        federationAdminService.updateRegistryEntry(updateMetacard);
+    }
+
+    @Override
+    public void deleteLocalEntry(List<String> ids) throws FederationAdminException {
+        if (CollectionUtils.isEmpty(ids)) {
+            throw new FederationAdminException(
+                    "Error deleting local registry entries. No ids provided.");
+        }
+
+        List<Metacard> localMetacards =
+                federationAdminService.getLocalRegistryMetacardsByRegistryIds(ids);
+        List<String> metacardIds = new ArrayList<>();
+
+        metacardIds.addAll(localMetacards.stream()
```

Comment:
The call to getLocalRegistryMetacardsByRegistryIds is to limit the delete to local metacards. Since the call returns metacards, using the metacard ids to delete.